sw/host/opentitanlib/src/transport/cw310/usb.rs

```diff
@@ -134,10 +137,10 @@ impl Backend {
     }
 
     /// Get the firmware version.
-    pub fn get_firmware_version(&self) -> Result<[u8; 3]> {
+    pub fn get_firmware_version(&self) -> Result<FirmwareVersion> {
         let mut buf = [0u8; 3];
         self.read_ctrl(Backend::CMD_FW_VERSION, 0, &mut buf)?;
-        Ok(buf)
+        Ok(FirmwareVersion(buf[0], buf[1], buf[2]))
```

Comment:
nit: I assume that you confirmed the endianness of this buffer.

Reply:
Yes, that's the main reason why `FirmwareVersion` derives `Debug` ;-)